http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/ProxyService.scala

```diff
@@ -115,6 +115,11 @@ class ProxyService(
           userInfo <- googleOauth2Service.getUserInfoFromToken(token)
           _ <-
             if (checkUserEnabled) for {
+              // TODO [] this code seems to assume a non-enabled user will return an error or an empty response. However,
```

Comment:
I'm trying to wrap my head around this logic. When do we call this with `checkUserEnabled == false`? And it looks like this code is just calling sam `samDAO.getSamUserInfo` and checking for a response but not actually checking the enabled flag. 🤯 

Maybe this code should use the new `samAuthProvider.checkUserEnabled` method. And investigate how we're using the `checkUserEnabled` flag.

Reply:
I created https://broadworkbench.atlassian.net/browse/IA-4131

Reply:
To answer your other Qs, we get user info "without" the enabled check (which isn't happening in any case but w/e) in ProxyRoutes:
```
          pathPrefix("proxy" | "notebooks") {

            corsSupport.corsHandler {

              refererHandler {
                // "apps" proxy routes
                pathPrefix("google" / "v1" / "apps") {
                  pathPrefix(googleProjectSegment / appNameSegment / serviceNameSegment) {
                    (googleProject, appName, serviceName) =>
                      extractUserInfoWithoutUserEnabledCheck(implicitly) { userInfo =>
```
and a couple other routes as well.